dcpquery/etl/__init__.py

```diff
@@ -255,3 +275,12 @@ def process_bundle_event(dss_event):
         drop_one_bundle(**dss_event["match"])
     else:
         logger.error("Ignoring unknown event type %s", dss_event["event_type"])
+
+
+@contextmanager
+def db_writes():
```

Comment:
Is this necessary? What are the circumstances under which readonly_db starts out as True but you want to force it to False? Also, I'm not sure if it does what you expect. `readonly_db` is only in effect for remote db, and I think you may have to flush the database connection before and after changing it.